account_asset_management_number/models/account_asset.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2021 Ecosoft Co., Ltd. (http://ecosoft.co.th)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class AccountAsset(models.Model):
+    _inherit = "account.asset"
+    _rec_name = "number"
+
+    number = fields.Char(
+        string="Asset Number",
+        default="",
+        index=True,
+        copy=False,
+    )
+    number_by_sequence = fields.Boolean(related="profile_id.number_by_sequence")
+
+    @api.model
+    def create(self, vals):
+        if vals.get("number", "") == "" and vals.get("profile_id"):
+            asset_profile = self.env["account.asset.profile"].browse(vals["profile_id"])
+            if asset_profile.number_by_sequence and asset_profile.sequence_id:
+                vals["number"] = asset_profile.sequence_id.next_by_id()
+        return super().create(vals)
+
+    def _insert_number_before_name(self, list_fields):
```

Comment:
So, I think this function is need necessary. WDYT?

Reply:
I already removed it.